src/stcal/jump/jump.py

```diff
@@ -340,50 +380,57 @@ def flag_large_events(gdq, jump_flag, sat_flag, min_sat_area=1,
+            current_gdq = 1.0 * gdq[integration, group, :, :]
+            prev_gdq = 1.0 * gdq[integration, group - 1, :, :]
+            diff_gdq = 1.0 * current_gdq - prev_gdq
+            diff_gdq[diff_gdq < 0] = 0
+            new_sat = diff_gdq.astype('uint8')
+            # find all the newly saturated pixel
+            sat_pixels = np.bitwise_and(diff_gdq.astype('uint8'), sat_flag)
+            saty, satx = np.where(sat_pixels == sat_flag)
+            only_jump = diff_gdq.copy()
+            # reset the saturated pixel to be jump to allow the jump circles to have the
+            # central saturated region set to "jump" instead of "saturation".
+            only_jump[saty, satx] = jump_flag
+            jump_ellipses = find_ellipses(only_jump.astype('uint8'), jump_flag, min_jump_area)
+            if sat_required_snowball:
+                low_threshold = edge_size
+                high_threshold = gdq.shape[2] - edge_size
+                snowballs = make_snowballs(jump_ellipses, sat_ellipses, low_threshold, high_threshold)
             else:
-                #  this line needs to be changed later to just look at saturation flags.
-                new_flagged_pixels = gdq[integration, group, :, :] - gdq[integration, group - 1, :, :]
-
-                sat_circles = find_circles(new_flagged_pixels, sat_flag, min_sat_area)
-                sat_pixels = np.bitwise_and(new_flagged_pixels, sat_flag)
-                saty, satx = np.where(sat_pixels == sat_flag)
-                only_jump = gdq[integration, group, :, :].copy()
-                only_jump[saty, satx] = jump_flag
-                jump_circles = find_circles(only_jump, jump_flag, min_jump_area)
-
-                if sat_required_snowball:
-                    snowballs = make_snowballs(jump_circles, sat_circles)
-                else:
-                    snowballs = jump_circles
-                n_showers_grp.append(len(snowballs))
-                gdq[integration, group, :, :], num_events = extend_snowballs(gdq[integration, group, :, :],
-                                                                             snowballs, sat_flag,
-                                                                             jump_flag,
-                                                                             expansion=expand_factor)
-        if use_ellipses:
+                snowballs = jump_ellipses
+            n_showers_grp.append(len(snowballs))
+            gdq, num_events = extend_ellipses(gdq, integration, group, snowballs, sat_flag,
+                                              jump_flag, expansion=expand_factor)
+        if find_faint_extended:
             if np.all(np.array(n_showers_grp_ellipse) == 0):
                 log.info(f'No showers found in integration {integration}.')
             else:
                 log.info(f' In integration {integration}, number of' +
                          f'showers in each group = {n_showers_grp_ellipse}')
```

Comment:
By the way, this isn't a change from the current code, but some of us have already noticed (since snowball was turned on a few days ago), that if you have an exposure with lots of integrations and/or lots of groups, these log messages can be really LARGE! (especially if `n_showers_grp_ellipse` is an array with many, many elements). Not sure if there's a more practical way to report this. Maybe the logging of per-group values could be changed to debug level, instead of info?

Reply:
The log message is only written once per exposure. That's what we had in the previous version too.